File: Taller 1/P2.Santiago.Diaz.py
# Santiago Díaz Obando, TI 1028481987

print("Bienvenido al ascensor del edificio.")

# Entrada y su verificación en el rango del edificio (Bucle hasta que se cumpla)
while True:
    floor = int(input("Introduzca en qué piso está: "))
    if floor >= 1 and floor <= 25:
        break
    else:
        print("El piso no es válido.")

# Añadir posibilidad de interacción mediante un bucle
while True:
    print(f"El ascensor está en el piso {floor}")

    # Lectura de posible acción
    action = input("Escriba subir, bajar o salir para realizar acciones: ").lower()

    # Verificar si es una acción válida y mover el ascensor
    if action == "subir":
        if floor < 25:
            print("Subiendo un piso...")
            floor += 1
        else:
            print("Este es el último piso!")
    elif action == "bajar":
        if floor > 1:
            print("Bajando un piso...")
            floor -= 1
        else:
            print("Este es el primer piso!")

    # Acción para romper el bucle
    elif action == "salir":
        print(f"Usted ha salido en el piso {floor} del edificio.")
        break

    # Caso por defecto
    else:
        print("Introduzca una acción válida.")
    
    print(" ")

File: Taller 1/P2.Santiago.Diaz.cpp
// Santiago Díaz Obando, TI 1028481987

#include <iostream>
#include <string>

using namespace std;

int main()
{
    // Declarar variables
    int f;
    string action;

    cout << "Bienvenido al ascensor del edificio." << endl;

    // Entrada del piso inicial
    cout << "Introduzca el piso en el que esta: ";
    cin >> f;

    // Verificar que sea un piso valido (1 - 25)
    if (f < 1 || f > 25) {
        cout << "No es un piso valido." << endl;
        return 0;
    }

    // Añadir posibilidad de interacción mediante un bucle
    while (true) {
        cout << "El ascensor está en el piso " << f << endl;

        // Entrada de una posible acciión del usuario
        cout << "Escriba subir, bajar o salir para realizar acciones: ";
        cin >> action;

        // Condicionales y acciones según entrada
        if (action == "subir") {
            if (f < 25) {
                cout << "Subiendo un piso..." << endl;
                f += 1;
            } else {
                cout << "Este es el último piso!" << endl;
            }
        }
        else if (action == "bajar") {
            if (f > 1) {
                cout << "Bajando un piso..." << endl;
                f -= 1;
            } else {
                cout << "Este es el primer piso!" << endl;
            }
        }
        // Si la acción es salir se rompe el ciclo
        else if (action == "salir") {
            cout << "Usted ha salido en el piso " << f << " del edificio." << endl;
            break;
        }
        // Caso por defecto
        else {
            cout << "No es una acción valida";
        }
        
    }
    
    return 0;
}